source/agora/node/Ledger.d

```diff
@@ -189,8 +206,12 @@ public class Ledger
 
     private const(Output)* findOutput (Hash tx_hash, size_t index) @safe
     {
-        foreach (ref block; this.ledger)
+        foreach (height; 0 .. this.block_height + 1)
         {
+            Block block;
+            if (this.ledger.readBlock(block, height))
```

Comment:
If `readBlock` succeeds, you break out of the foreach loop and you return null. So the output will never be found. I think you meant to do `if (!this.ledger.readBlock(block, height)) return null`.

Reply:
Why am I taking this away? Thank you for finding me.